src/main/java/com/squareup/catalog/demo/example/clone/CatalogObjectCloneUtil.java

```diff
@@ -0,0 +1,112 @@
+   * CatalogObject} in the target account, then the object is not cloned.
+   */
+  final String encodeCatalogObject(CatalogObject catalogObject) {
+    T catalogData = getCatalogData(catalogObject);
+    return encodeCatalogData(catalogData);
+  }
+
+  /**
+   * Encodes a subset of fields from the {@link CatalogObject} into a string, allowing
+   * CatalogObjects to be compared between merchant accounts for likeness. If the encoded string of
+   * a {@link CatalogObject} in the source account matches the encoded string of a {@link
+   * CatalogObject} in the target account, then the object is not cloned.
+   *
+   * @param catalogData the data from the {@link CatalogObject}.
+   */
+  abstract String encodeCatalogData(T catalogData);
+
+  /**
+   * Encodes a {@link Money} object to a string.
+   *
+   * @param money the {@link Money} object to encode
+   * @return the amount as a string, or null if the {@link Money} object is null
+   */
+  String amountOrNull(Money money) {
+    return (money == null) ? "null" : money.getAmount().toString();
+  }
+
+  /**
+   * Removes meta data from the {@link CatalogObject} that only applies to the source account, such
+   * as location IDs and version.
+   *
+   * @param catalogObject the {@link CatalogObject} to modify
+   */
+  void removeSourceAccountMetaData(CatalogObject catalogObject) {
+    // We need to set a temporary client generated ID.
+    catalogObject.setId("#" + UUID.randomUUID());
+
+    // The V1 IDs from the source account do not apply to the target account.
+    catalogObject.setCatalogV1Ids(Collections.emptyList());
+
+    // The location IDs from the source account do not apply to the target account.
+    catalogObject.setPresentAtLocationIds(Collections.emptyList());
+    catalogObject.setAbsentAtLocationIds(Collections.emptyList());
+
+    // The server will assign a version.
+    catalogObject.setVersion(null);
+
+    // The server will update the timestamp.
+    catalogObject.setUpdatedAt(null);
+  }
+
+  /**
+   * Merges data from a {@link CatalogObject} from the source account into a {@link CatalogObject}
+   * from the target account when they have the same encoded string. This method can be used to
+   * merge two objects that have the same name, but slightly different attributes. For example, if a
+   * modifier list in the source account has the same name as a modifier list in the target account,
+   * this method is used to copy the modifiers from the source account to the target account.
+   *
+   * @param sourceCatalogObject the {@link CatalogObject} from the source account
+   * @param targetCatalogObject the {@link CatalogObject} from the target account
+   * @return the modifier target {@link CatalogObject}, or null if nothing changed
+   */
+  CatalogObject mergeSourceCatalogObjectIntoTarget(CatalogObject sourceCatalogObject,
+      CatalogObject targetCatalogObject) {
+    // Default implementation assumes that the objects do not need to be merged.
+    return null;
```

Comment:
No, by default we assume that there is nothing to merge. Generally we only need to merge if the object contains child objects (ex. modifier list contains modifiers).